tidy3d/plugins/design/method.py

```diff
@@ -23,335 +20,780 @@ class Method(Tidy3dBaseModel, ABC):
+    )
 
-        d = len(parameters)
-        return DEFAULT_MONTE_CARLO_SAMPLER_TYPE(d=d)
+    n_iter: pd.PositiveInt = pd.Field(
+        ...,
+        title="Number of bayesian optimization iterations",
+        description="Following the initial search, this is number of iterations the Gaussian processor should be sequentially called to suggest parameter values and register the results.",
+    )
 
+    acq_func: Literal["ucb", "ei", "poi"] = pd.Field(
+        default="ucb",
+        title="Type of acquisition function",
+        description="The type of acquisition function that should be used to suggest parameter values. More detail available `here <https://bayesian-optimization.github.io/BayesianOptimization/exploitation_vs_exploration.html>`_",
+    )
 
-class MethodRandom(AbstractMethodRandom):
-    """Select sampling points uniformly at random.
+    kappa: pd.PositiveFloat = pd.Field(
+        default=2.5,
+        title="Kappa",
+        description="The kappa coefficient used by the ``ucb`` acquisition function. More detail available `here <https://bayesian-optimization.github.io/BayesianOptimization/exploitation_vs_exploration.html>`_",
+    )
 
-    Example
-    -------
-    >>> import tidy3d.plugins.design as tdd
-    >>> method = tdd.MethodRandom(num_points=20, monte_carlo_warning=False)
+    xi: pd.NonNegativeFloat = pd.Field(
+        default=0.0,
+        title="Xi",
+        description="The Xi coefficient used by the ``ei`` and ``poi`` acquisition functions. More detail available `here <https://bayesian-optimization.github.io/BayesianOptimization/exploitation_vs_exploration.html>`_",
+    )
+
+    def get_run_count(self, parameters: list = None) -> int:
+        return self.initial_iter + self.n_iter
+
+    def _run(self, parameters: Tuple[ParameterType, ...], run_fn: Callable, console) -> Tuple[Any]:
+        """Defines the bayesian optimization search algorithm for the method.
+
+        Uses the ``bayes_opt`` package to carry out a Bayesian optimization. Utilizes the ``.suggest`` and ``.register`` methods instead of
+        the ``BayesianOptimization`` helper class as this allows more control over batching and preprocessing.
+        More details of the package can be found `here <https://bayesian-optimization.github.io/BayesianOptimization/basic-tour.html>'_
+        """
+        try:
+            from bayes_opt import BayesianOptimization, UtilityFunction
+        except ImportError:
+            raise ImportError(
+                "Cannot run bayesian optimization as 'bayes_opt' module not found. Please check installation or run 'pip install bayesian-optimization'."
+            )
+
+        param_converter = {}
+        boundary_dict = {}
+        for param in parameters:
+            if type(param) == ParameterAny:
```

Comment:
`isinstance(param, ParameterAny)`, in case we subclass ParameterAny down the line